### ISSUE TITLE
Support `xser` for saving checkpoints for XLA

### ISSUE THREAD

**Issue by kaushikb11** [PyTorchLightning/pytorch-lightning#7969]
## 🚀 Feature

```python
import torch_xla.utils.serialization as xser

xser.save(model.state_dict(), path)
```

### Motivation

In case where memory is limited compared to the size of the model parameters, an API is provided that reduces the memory footprint on the host:

```python
import torch_xla.utils.serialization as xser

xser.save(model.state_dict(), path)
```

This API streams XLA tensors to CPU one at a time, reducing the amount of host memory used, but it requires a matching load API to restore:

```python
import torch_xla.utils.serialization as xser

state_dict = xser.load(path)
model.load_state_dict(state_dict)
```

**Comment by tchaton** [comment]
Hey @kaushikb11,

Should we make this an argument to the TPUSpawn Plugin and use it by default ?

Best,
T.C

**Comment by kaushikb11** [comment]
Yup, having it as an argument is what I had in mind. Planning to work on it tonight.

**Comment by carmocca** [comment]
This is a low-hanging fruit. Opened https://github.com/pytorch/xla/pull/4899 to expose the map location
